### ISSUE TITLE
Implement `Function::Custom` for custom storage developer

### ISSUE THREAD

**Issue by ever0de** [gluesql/gluesql#660]
We found a need to easily create custom functions in the references below.
- <https://github.com/KOBA789/gh-sql/issues/12>

Function parsing in `sqlparser-rs` crate is already supported now.
https://github.com/sqlparser-rs/sqlparser-rs/blob/f74753436a57c961a17863168b84056847f88edf/src/ast/mod.rs#L1039-L1044

### Example

If it is a function name that is not supported by `translate`, an error is returned. There is a way to pass it to Custom and use it if the name is in the store trait for the custom function implemented by the user.

```rust
/// declare `function` enum variant
Function::Custom {
    name: String,
    args: Vec<Expr>
}

// The name has not been decided.
trait CustomFunction {
    fn custom_function(self, name: String, args: Vec<Expr>) -> Result<Evaluated>
}
```

- Why the result value is `Result<Evaluated>`

https://github.com/gluesql/gluesql/blob/3551c6f6e9d395aa3fd884e9230582c430f0f5d7/core/src/executor/evaluate/mod.rs#L203-L208

**Comment by pythonbrad** [comment]
Hi, i tried to solve this issue. I think that the usage of trait is not the good idea. It required to enable public access to some modules.

**Comment by pythonbrad** [comment]
@ever0de , what about this method ?
- [hibernate-sql-function-jpql-criteria-api-query](https://vladmihalcea.com/hibernate-sql-function-jpql-criteria-api-query)
- [registering-a-sql-function-with-jpa-and-hibernate](https://stackoverflow.com/questions/12346845/registering-a-sql-function-with-jpa-and-hibernate)

**Comment by ever0de** [comment]
Adding an additional meta builder would also be nice.
But I'm still not sure at what point I should put 😅 

Since we also need to support user-level custom functions, how about implementing this first? (Create an issue separate from that issue)

**Comment by pythonbrad** [comment]
> Since we also need to support user-level custom functions, how about implementing this first? 

I will make a pull request

**Comment by panarch** [comment]
resolved in #1095
